sktime/utils/validation/__init__.py

```diff
@@ -83,6 +83,36 @@ def array_is_timedelta_or_date_offset(x) -> bool:
     return all([is_timedelta_or_date_offset(value) for value in x])
 
 
+def is_iloc_like(x) -> bool:
+    """Check if input is .iloc friendly."""
+    try:
+        iter(x)
```

Comment:
I first thought this is really strange, but a quick investigation seems to confirm that this is one of the most reliable ways to check whether an object is iterable...

Either way, this occurs two times, so I´d recommend to factor this check out in a function `is_iterable`.

Reply:
done, pushed